inst/include/dust/rng.hpp

```diff
@@ -80,6 +81,30 @@ class pRNG { // # nocov
   std::vector<rng_state_t<T>> _state;
 };
 
+template <typename T>
+rng_state_t<T> get_rng_state(const dust::interleaved<uint64_t>& full_rng_state) {
+  rng_state_t<T> rng_state;
+  for (size_t i = 0; i < rng_state_t<T>::size(); i++) {
+    rng_state.state[i] = full_rng_state[i];
+  }
+  return rng_state;
+}
+
+// Write state into global memory
+template <typename T>
+void put_rng_state(rng_state_t<T>& rng_state,
+                   dust::interleaved<uint64_t>& full_rng_state) {
+  for (size_t i = 0; i < rng_state_t<T>::size(); i++) {
+    full_rng_state[i] = rng_state.state[i];
+  }
+}
+
 }
 
 #endif
+
+template <typename T, typename U = typename T::real_t>
+U unif_rand(T& state) {
+  const uint64_t value = xoshiro_next(state);
+  return U(value) / U(std::numeric_limits<uint64_t>::max());
```

Comment:
I liked your idea about the numeric limits helper class. Shall we note this down somewhere?

Reply:
Added here: https://github.com/mrc-ide/dust/issues/165